src/tracing/on_disk.rs

```diff
@@ -0,0 +1,95 @@
+use std::collections::HashSet;
+use std::fs;
+
+use anyhow::Context as _;
+use serde::{Deserialize, Serialize};
+use tracing_subscriber::prelude::*;
+use tracing_subscriber::{filter, fmt, registry};
+
+use super::*;
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(default)]
+pub struct Config {
+    pub enabled: Option<bool>,
+    pub log_file: Option<String>,
+    pub log_level: Option<String>,
+    pub span_events: Option<HashSet<config::SpanEvent>>,
+}
```

Comment:
When fetching `GET /logger` I get a default structure like this:

```json
{
  "log_level": "DEBUG",
  "span_events": null,
  "color": null,
  "on_disk": {
    "enabled": null,
    "log_file": null,
    "log_level": null,
    "span_events": null
  }
}
```

Wouldn't it be much nicer to enable/disable on-disk logging based on whether the configuration is available?

Disabled:

```json
{
  "log_level": "DEBUG",
  "span_events": null,
  "color": null,
  "on_disk": null
}
```

Enabled:

```json
{
  "log_level": "DEBUG",
  "span_events": null,
  "color": null,
  "on_disk": {
    "log_file": "qdrant.log",
    "log_level": null,
    "span_events": null
  }
}
```

Or is there a reason we cannot do this?

Reply:
Then you'd have to:
- provide the full `on_disk` object every time you want to enable on-disk logging
- you can't have on-disk logging pre-configured (e.g., log-file, log-level, etc) in the config file, **but** disabled

Reply:
Thanks for elaborating!